tarantool/connection.py

```diff
@@ -298,6 +329,46 @@ def connect_unix(self):
             self.connected = False
             raise NetworkError(e)
 
+    def wrap_socket_ssl(self):
+        '''
+        Wrap an existing socket with SSL socket.
+
+        :raise: NetworkError
+        :raise: `ssl.SSLError`
+        '''
+        if is_ssl_supported == False:
+            raise NetworkError("SSL is unsupported by the python.")
+
+        if hasattr(ssl, 'TLSVersion'):
+            # Since python 3.7
+            context = ssl.create_default_context(ssl.Purpose.SERVER_AUTH)
+            # Reset to default OpenSSL values
+            context.check_hostname = False
+            context.verify_mode = ssl.CERT_NONE
+            # Require TLSv1.2, because other protocol versions don't seem to
+            # support the GOST cipher.
+            context.minimum_version = ssl.TLSVersion.TLSv1_2
+            context.maximum_version = ssl.TLSVersion.TLSv1_2
+        else:
+            # Deprecated, but it works for python < 3.7
+            context = ssl.SSLContext(ssl.PROTOCOL_TLSv1_2)
+
+        if self.ssl_cert_file:
+            context.load_cert_chain(certfile = self.ssl_cert_file,
+                                    keyfile = self.ssl_key_file)
```

Comment:
```
FAIL: test_single (suites.test_ssl.TestSuite_Ssl) [key_crt_ca_server_and_crt_client]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ozi/src/git/tarantool-python/tarantool/connection.py", line 393, in connect
    self.wrap_socket_ssl()
  File "/home/ozi/src/git/tarantool-python/tarantool/connection.py", line 356, in wrap_socket_ssl
    context.load_cert_chain(certfile=self.ssl_cert_file,
ssl.SSLError: [SSL] PEM lib (_ssl.c:3895)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ozi/src/git/tarantool-python/test/suites/test_ssl.py", line 251, in test_single
    con = tarantool.Connection(
  File "/home/ozi/src/git/tarantool-python/tarantool/connection.py", line 269, in __init__
    self.connect()
  File "/home/ozi/src/git/tarantool-python/tarantool/connection.py", line 398, in connect
    raise NetworkError(e)
tarantool.error.NetworkError: [SSL] PEM lib (_ssl.c:3895)
```
[But it may be a valid configuration:](https://docs.python.org/3/library/ssl.html#ssl.SSLContext.load_cert_chain)
```
Load a private key and the corresponding certificate. The certfile string must be the path to a single file in PEM format containing the certificate as well as any number of CA certificates needed to establish the certificate’s authenticity. The keyfile string, if present, must point to a file containing the private key. Otherwise the private key will be taken from certfile as well. See the discussion of [Certificates](https://docs.python.org/3/library/ssl.html#ssl-certificates) for more information on how the certificate is stored in the certfile.
```

Reply:
But from our docs it [both mandatory](https://webcache.googleusercontent.com/search?q=cache:hMW4x7zmT6EJ:https://www.tarantool.io/en/enterprise_doc/security/+&cd=2&hl=ru&ct=clnk&gl=ru) ssl_ca_file is set on a server-side, so I'll change it too:
```
tarantool.error.NetworkError: ssl_cert_file and ssl_key_file should be  both configured or not
```

Reply:
[From the other side](https://webcache.googleusercontent.com/search?q=cache:hMW4x7zmT6EJ:https://www.tarantool.io/en/enterprise_doc/security/+&cd=2&hl=ru&ct=clnk&gl=ru) it's both ssl_cert_file and ssl_key_file mandatory in the Tarantool's reference implementation, so it may reason to change the error too:
```
ssl_cert_file and ssl_key_file should be both configured or not
```
But it seems to me that in the Python implementation we can do not such a restriction. And leave as it is.

Reply:
I'm okay to don't restrict it if we'll give an error suitable to understand the problem.

Reply:
From this point of view I prefer just check the values and raise if needed:
```ssl_cert_file and ssl_key_file should be both configured or not```

I can catch an exception from the `ssl` module and process to make it more understandable, but it may be difficult to support in the future.